pallets/ajuna-awesome-avatars/src/lib.rs

```diff
@@ -392,15 +395,13 @@ pub mod pallet {
 		}
 
 		pub(crate) fn random_dna(
-			who: &T::AccountId,
+			hash: &T::Hash,
 			season: &SeasonOf<T>,
 		) -> Result<(Dna, bool), DispatchError> {
-			let hash = Self::random_hash(who);
 			let dna = (0..season.max_components)
-				.map(|base_index| {
-					let (random_tier, random_variation) =
-						Self::random_component(season, &hash, base_index as usize * 2);
-					((random_tier << 4) | random_variation) as u8
+				.map(|i| {
+					let (tier, variation) = Self::random_component(season, hash, i as usize * 2);
+					((tier << 4) | variation) as u8
```

Comment:
- Changed to just `i` because `base_index` was a bit confusing to me :sweat_smile:
- Also renamed variables to fit in one line for readability